fs/reader/reader.go

```diff
@@ -92,8 +94,20 @@ func (nv nopVerifier) Verified() bool {
 // NewReader creates a Reader based on the given stargz blob and cache implementation.
 // It returns VerifiableReader so the caller must provide a estargz.TOCEntryVerifier
 // to use for verifying file or chunk contained in this stargz blob.
-func NewReader(sr *io.SectionReader, cache cache.BlobCache) (*VerifiableReader, error) {
-	r, err := estargz.Open(sr)
+func NewReader(sr *io.SectionReader, cache cache.BlobCache, layerDigest digest.Digest) (*VerifiableReader, error) {
```

Comment:
Could we pass `*estargz.Telemetry` to `NewReader()` instead of `digest.Digest`? It would be great if we can reuse this `*estargz.Telemetry` also in `reader.Cache()`.

Reply:
That works. Will update in the next revision of the PR.

Reply:
All set.